### ISSUE TITLE
Some linter errors not reported

### ISSUE THREAD

**Issue by petobens** [jimhester/lintr#315]
Hi, I recently discovered (and loved) lintr. There are few things I can't however seem to understand.
Consider the following code:
```R
a_very_long_var = 2
x <<- 5
k 
```
When I run lintr with the following settings:
```
with_defaults(
    line_length_linter(80), 
    commented_code_linter = NULL, 
    camel_case_linter,
    snake_case_linter = NULL
)
```
The only message I get is:
```
Use <-, not =, for assignment.
```
So: i) why are the `undesirable_operation_linter` warnings not shown (since I used it in `x <<- 5`_? ii) why there is no  warning about unused variables (for instance for `a_very_long_var`) and iii) if I run this code then `k` is clearly an error: why is this not shown?

Thanks!!

**Comment by jimhester** [comment]
i) `undesirable_operation_linter` is not in the default linters.
ii, iii) There would be warnings if this code was in a function, this is just due to how lintr works.

If I edit your script to be in a function this is what I get

```r
f <- function() {
  a_very_long_var = 2
  x <<- 5
  k 
}
```

```r
lint("script.R", linters = with_defaults(undesirable_operator_linter()))
#> script.R:2:3: warning: local variable
#>  ‘a_very_long_var’ assigned but may not be used
#>   a_very_long_var = 2
#>   ^~~~~~~~~~~~~~~
#> script.R:2:19: style: Use <-, not =,
#> for assignment.
#>   a_very_long_var = 2
#>                   ^
#> script.R:3:3: warning: no visible bin
#> ding for '<<-' assignment to ‘x’, Did you mean ':'?
#>   x <<- 5
#>   ^
#> script.R:3:5: warning: Operator `<<-`
#>  is undesirable.
#>   x <<- 5
#>     ^~~
#> script.R:4:3: warning: no visible bin
#> ding for global variable ‘k’, Did you mean ':'?
#>   k
#>   ^
#> script.R:4:4: style: Trailing whitesp
#> ace is superfluous.
#>   k
#>    ^
```

I think this is all working as intended, so I am going to close this issue.
